src/sgx/rust_sgx_bridge/rust_sgx_bridge.cpp

```diff
@@ -0,0 +1,177 @@
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+#include "poet_enclave_u.h"
+#include "sgx_urts.h"
+#include "rust_sgx_bridge.h"
+#include "poet_enclave.h"
+#include "c11_support.h"
+#include "error.h"
+#include <iostream>
+#include "poet.h"
+
+Poet* Poet::instance = 0;
+
+WaitCertificate* validate_wait_certificate(const char* ser_wait_cert, 
+              const char* ser_wait_cert_sig);
+
+int r_initialize_enclave(r_sgx_enclave_id_t *eid, const char * enclave_path, 
+                         const char *spid) 
+{
+    //check parameters
+    if (!eid || !enclave_path || !spid) {
+        return -1;
+    }
+    try{
+        Poet *poet_enclave_id = Poet::getInstance(enclave_path, spid);
+        //store the enclave id
+        eid->handle = (intptr_t)poet_enclave_id;
+    } catch( sawtooth::poet::PoetError& e) {   
+        return -1;
+    }
+    return 0;
+}
+
+int r_free_enclave(r_sgx_enclave_id_t *eid)
+{
+    if (eid->handle != 0) {
+      Poet *poet_enclave_id = (Poet *)eid->handle;   
+      delete poet_enclave_id;   
+      eid->handle = 0;
+    }
+
+    return 0;
+}
+
+int r_create_signup_info(r_sgx_enclave_id_t *eid, const char *opk_hash, 
+                         r_sgx_signup_info_t *signup_info) {
+  if (!eid || !opk_hash || !signup_info) {
+        return -1;
+  }
+  if (eid->handle == 0) {
+       return -1;
+  }
+  _SignupData *signup_data = _create_signup_data(opk_hash);
+  if (signup_data == NULL) {
+       return -1;
+  }
+   //store the _SignupData handle
+   signup_info->handle = (intptr_t)signup_data; 
+   if (signup_data->poet_public_key.empty()) {
+      return -1;
+   }
+   signup_info->poet_public_key = (char *)signup_data->poet_public_key.c_str();
+   if (signup_data->sealed_signup_data.empty()) {
+      return -1;
+   }
+   signup_info->sealed_signup_data = (char *)signup_data->sealed_signup_data.c_str();    
+   return 0;   
+}
+
+
+int r_initialize_wait_certificate(r_sgx_enclave_id_t *eid, uint8_t* duration, 
+                                  const char* prevCert, const char* prevBlockId, 
+                                  const char* validatorId) {
+    if (!eid || (prevCert == NULL) || (prevBlockId == NULL) || (validatorId == NULL)) {
+       return -1;
+    }
+    if (eid-> handle == 0) {
+        return -1;
+    }
+    poet_err_t ret = initialize_wait_certificate(prevCert, prevBlockId, validatorId, 
+                                                duration, 32); 
```

Comment:
The no. 32 could be dnoted in a more meningful manner in a const variable or #define